metaseq/tasks/streaming_language_modeling.py

```diff
@@ -308,13 +360,37 @@ def load_dataset(self, split: str, epoch=1, combine=False, **kwargs):
                 )
             )
             corpora.append(os.path.splitext(file)[0])
+
+            dataset_name_to_index[os.path.join(self.args.data, split, cur_shard_str, file)] = dataset_index_counter
+            dataset_index_counter += 1
+
         assert len(datasets) > 0
 
         if self.args.multicorpus_sampling_alpha != 1:
             datasets = self._alpha_sampling(datasets, corpora, epoch)
 
         dataset = torch.utils.data.ConcatDataset(datasets)
 
+        # Filter by metric (only on train dataset)
```

Comment:
I put the filter dataset here instead of later on, because there's no point in processing on data points we are sure that we are not going to use (i.e shuffling, or dividing into token blocks);

however, we want to filter after concat because I thought it would be cleaner to apply the filter one time on the entire original dataset (instead of applying a filter multiple times on portions of the dataset i.e on the individual `JsonlDataset`'s)